compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -125,6 +135,19 @@ PackageContext packageContext() {
         return topologicallySortedModuleList;
     }
 
+    List<Diagnostic> diagnostics() {
+        return this.diagnosticList;
```

Comment:
We return the DiagnosticResult in other phases. Shall we follow the same pattern here?

Reply:
Added